process/rating/ratingsData.go

```diff
@@ -114,66 +117,100 @@ func NewRatingsData(args RatingsDataArg) (*RatingsData, error) {
+	if err != nil {
+		return nil, err
+	}
+
 	args.EpochNotifier.RegisterNotifyHandler(ratingData)
 
 	return ratingData, nil
 }
 
+func (rd *RatingsData) computeRatingStepDataMaps(chainParamsList []config.ChainParametersByEpochConfig) error {
+	shardRatingsSteps := make(map[uint32]process.RatingsStepHandler)
+	metaRatingsSteps := make(map[uint32]process.RatingsStepHandler)
+	for _, chainParams := range chainParamsList {
+		shardRatingsStepsArgs := computeRatingStepArg{
+			shardSize:                       chainParams.ShardMinNumNodes,
+			consensusSize:                   chainParams.ShardConsensusGroupSize,
+			roundTimeMillis:                 rd.roundDurationInMilliseconds,
+			startRating:                     rd.ratingsSetup.General.StartRating,
+			maxRating:                       rd.ratingsSetup.General.MaxRating,
+			hoursToMaxRatingFromStartRating: rd.ratingsSetup.ShardChain.HoursToMaxRatingFromStartRating,
+			proposerDecreaseFactor:          rd.ratingsSetup.ShardChain.ProposerDecreaseFactor,
+			validatorDecreaseFactor:         rd.ratingsSetup.ShardChain.ValidatorDecreaseFactor,
+			consecutiveMissedBlocksPenalty:  rd.ratingsSetup.ShardChain.ConsecutiveMissedBlocksPenalty,
+			proposerValidatorImportance:     rd.ratingsSetup.ShardChain.ProposerValidatorImportance,
+		}
+		shardRatingsStepData, err := computeRatingStep(shardRatingsStepsArgs)
+		if err != nil {
+			return fmt.Errorf("%w while computing shard rating steps for epoch %d", err, chainParams.EnableEpoch)
+		}
+		shardRatingsSteps[chainParams.EnableEpoch] = shardRatingsStepData
+
+		metaRatingsStepsArgs := computeRatingStepArg{
+			shardSize:                       chainParams.MetachainMinNumNodes,
+			consensusSize:                   chainParams.MetachainConsensusGroupSize,
+			roundTimeMillis:                 rd.roundDurationInMilliseconds,
+			startRating:                     rd.ratingsSetup.General.StartRating,
+			maxRating:                       rd.ratingsSetup.General.MaxRating,
+			hoursToMaxRatingFromStartRating: rd.ratingsSetup.MetaChain.HoursToMaxRatingFromStartRating,
+			proposerDecreaseFactor:          rd.ratingsSetup.MetaChain.ProposerDecreaseFactor,
+			validatorDecreaseFactor:         rd.ratingsSetup.MetaChain.ValidatorDecreaseFactor,
+			consecutiveMissedBlocksPenalty:  rd.ratingsSetup.MetaChain.ConsecutiveMissedBlocksPenalty,
+			proposerValidatorImportance:     rd.ratingsSetup.MetaChain.ProposerValidatorImportance,
+		}
+		metaRatingsStepData, err := computeRatingStep(metaRatingsStepsArgs)
+		if err != nil {
+			return fmt.Errorf("%w while computing metachain rating steps for epoch %d", err, chainParams.EnableEpoch)
+		}
+		metaRatingsSteps[chainParams.EnableEpoch] = metaRatingsStepData
+	}
+
+	rd.shardRatingsStepsDataValues = shardRatingsSteps
+	rd.metaRatingsStepsDataValues = metaRatingsSteps
+
+	return nil
+}
+
 // EpochConfirmed will be called whenever a new epoch is called
 func (rd *RatingsData) EpochConfirmed(epoch uint32, _ uint64) {
```

Comment:
we need here a "getMatchingConfig..." function.
Reason:
this doesn't work as expected for the following scenario:
- suppose all the chain nodes started from the genesis block, iterating through several chain changes up until epoch `e`
- a nod starts in epoch `e` directly. 
- if the epoch `e`  does not contain a change in the map, the node starts with a default set of values and is not able to sync the chain

Reply:
good catch. refactored to use a slice of element (so that it can be sorted ascending based on epochs)